addons/sale_project/models/project.py

```diff
@@ -646,51 +648,68 @@ def _get_revenues_items_from_invoices(self, excluded_move_line_ids=None, with_ac
+                move_ids = set()
+                amount_invoiced = amount_to_invoice = 0.0
+                for moves_read in moves:
+                    currency = self.env['res.currency'].browse(moves_read['currency_id']).with_prefetch(currency_ids)
+                    price_subtotal = currency._convert(moves_read['price_subtotal'], self.currency_id, self.company_id)
+                    # an analytic account can appear several time in an analytic distribution with different repartition percentage
+                    analytic_contribution = sum(
+                        percentage for ids, percentage in moves_read['analytic_distribution'].items()
+                        if str(self.analytic_account_id.id) in ids.split(',')
+                    ) / 100.
+                    move_ids.add(moves_read['move_id'])
+                    if moves_read['parent_state'] == 'draft':
+                        if moves_read['move_type'] == 'out_invoice':
+                            amount_to_invoice += price_subtotal * analytic_contribution
+                        else:  # moves_read['move_type'] == 'out_refund'
+                            amount_to_invoice -= price_subtotal * analytic_contribution
+                    else:  # moves_read['parent_state'] == 'posted'
+                        if moves_read['move_type'] == 'out_invoice':
+                            amount_invoiced += price_subtotal * analytic_contribution
+                        else:  # moves_read['move_type'] == 'out_refund'
+                            amount_invoiced -= price_subtotal * analytic_contribution
+                # don't display the section if the final values are both 0 (invoice -> credit note)
+                if amount_invoiced != 0 or amount_to_invoice != 0:
+                    section_id = 'other_invoice_revenues' if ml_type == 'revenues' else 'cost_of_goods_sold'
+                    invoices_items = {
+                        'id': section_id,
+                        'sequence': self._get_profitability_sequence_per_invoice_type()[section_id],
+                        'invoiced' if ml_type == 'revenues' else 'billed': amount_invoiced,
+                        'to_invoice' if ml_type == 'revenues' else 'to_bill': amount_to_invoice,
+                    }
+                    if with_action and self.user_has_groups('sales_team.group_sale_salesman_all_leads,account.group_account_invoice,account.group_account_readonly'):
+                        invoices_items['action'] = self._get_action_for_profitability_section(list(move_ids), section_id)
+                    res[ml_type] = {
+                        'data': [invoices_items],
+                        'total': {
+                            'invoiced' if ml_type == 'revenues' else 'billed': amount_invoiced,
+                            'to_invoice' if ml_type == 'revenues' else 'to_bill': amount_to_invoice,
+                        },
+                    }
+            return res
+        return res
```

Comment:
I mean why not doing that: 
```suggestion
        return res
```
? 🤔

Reply:
Good question, clearly you're right :sweat_smile: